Week3/homework/step2-3.js

```diff
@@ -4,23 +4,28 @@
 function repeatStringNumTimesWithFor(str, num) {
   // eslint-disable-next-line prefer-const
   let result = '';
-
-  // Replace this comment and the next line with your code
-  console.log(str, num, result);
-
+  if (num <= 0) {
+    return result;
+  }
+  for (let i = 0; i < num; i++) {
+    result += str;
+  }
   return result;
 }
 
 console.log('for', repeatStringNumTimesWithFor('abc', 3));
 
 // Use a 'while' loop
 function repeatStringNumTimesWithWhile(str, num) {
-  // eslint-disable-next-line prefer-const
   let result = '';
-
-  // Replace this comment and the next line with your code
-  console.log(str, num, result);
-
+  if (num <= 0) {
```

Comment:
While this is a `while` loop and line 7 you have a `for` loop, checking for `num` if it's less than 0 would be a redundant step, the for/while loops will not loop through what you have unless the condition is correct, so:
if `i < num` and `i` is 0 the condition will never pass

the only one needs that condition is the `do while` loop because it's doing before it checks!